types/tough-cookie-file-store/index.d.ts

```diff
@@ -31,7 +31,8 @@ export class FileCookieStore extends tough.Store {
     findCookies(
         domain: string,
         path: string,
-        cb: (err: null, cookies: tough.Cookie[]) => void
+        allowSpecialUseDomain: boolean,
+        cb: (err: Error | null, cookie: tough.Cookie[]) => void
```

Comment:
This should add a new overload signature, not replace the existing one

Reply:
According to the types defined in the [```@types/tough-cookie```](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/tough-cookie/index.d.ts) module the argument "allowSpecialUseDomain" is not optional for findCookies and there are no overloads specified.
This is reflected in the module [```tough-cookie-file-store```](https://github.com/ivanmarban/tough-cookie-file-store/blob/master/lib/cookie-file-store.js) which follows the same structure without any overload.

From what I understand, adding a overload in the types would allow for incorrect use of findCookies resulting in no compile time errors but runtime errors if someone was to attempt to call it without passing the argument "allowSpecialUseDomain".

So the current changes should be used rather than a overload right?
If it is a matter of backwards compatibility then perhaps the same method [```@types/tough-cookie```](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/tough-cookie) currently uses, where there is a ```v2``` folder with seperate types defined for use with that version of tough-cookie should be used.

Let me know your thoughts.

Reply:
Looking over the types [```@types/tough-cookie-file-store```](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/tough-cookie-file-store) and library [```tough-cookie-file-store```](https://github.com/ivanmarban/tough-cookie-file-store) it appears that actually quite a lot has changed and many functions such as **```isEmpty```** no longer exist. 

If you agree with the suggestion of moving the current (pre these two commits) state of the types to a v2 folder, I would be happy to go over all the types and update them so they are in line with the current state of the [```tough-cookie-file-store```](https://github.com/ivanmarban/tough-cookie-file-store) module, as currently they arent.

Reply:
Looking at the pull request you linked, it supports an overload -- `allowSpecialUseDomain` _is_ optional. https://github.com/ivanmarban/tough-cookie-file-store/pull/8/files#diff-21173867e25ead79fef51ae270700bd02957349c9b67b8bd4af29afc6fd89035R84 https://github.com/ivanmarban/tough-cookie-file-store/blob/master/lib/cookie-file-store.js#L92

Reply:
And yes - since these types are for v1, v2 should be in a different folder as it has breaking changes

Reply:
> Looking at the pull request you linked, it supports an overload -- `allowSpecialUseDomain` _is_ optional. https://github.com/ivanmarban/tough-cookie-file-store/pull/8/files#diff-21173867e25ead79fef51ae270700bd02957349c9b67b8bd4af29afc6fd89035R84 https://github.com/ivanmarban/tough-cookie-file-store/blob/master/lib/cookie-file-store.js#L92

Ah yes. I missed that, thankyou. I'll add a overload.